File: game_plugin_interface/src/sc/protocol/requests/AuthenticateRequest.java
package sc.protocol.requests;

import com.thoughtworks.xstream.annotations.XStreamAlias;
import com.thoughtworks.xstream.annotations.XStreamAsAttribute;

@XStreamAlias("authenticate")
public class AuthenticateRequest implements ILobbyRequest
{
	@XStreamAsAttribute
	private String	password;

	public AuthenticateRequest(String password)
	{
		this.password = password;
	}

	public String getPassword()
	{
		return this.password;
	}
}

File: sdk/src/server-api/sc/protocol/requests/AuthenticateRequest.kt
package sc.protocol.requests

import com.thoughtworks.xstream.annotations.XStreamAlias
import com.thoughtworks.xstream.annotations.XStreamAsAttribute

/** Authenticates a client as administrator to send [AdminLobbyRequest]s.
 * Is not answered if successful. */
@XStreamAlias("authenticate")
data class AuthenticateRequest(
        @XStreamAsAttribute
        val password: String
): ILobbyRequest
